src/Coordination/KeeperServer.h

```diff
@@ -24,15 +25,19 @@ class KeeperServer
 
     nuraft::ptr<KeeperStateManager> state_manager;
 
-    nuraft::ptr<nuraft::raft_server> raft_instance;
+    struct KeeperRaftServer;
+    nuraft::ptr<KeeperRaftServer> raft_instance;
     nuraft::ptr<nuraft::asio_service> asio_service;
     nuraft::ptr<nuraft::rpc_listener> asio_listener;
+    mutable std::mutex server_mutex;
```

Comment:
Not clear what is guarded with this mutex and how we should use it. Add a comment.